flang/include/flang/Lower/CharacterExpr.h

```diff
@@ -101,6 +101,10 @@ class CharacterExprHelper {
   std::pair<mlir::Value, mlir::Value>
   materializeCharacterOrSequence(mlir::Value str);
 
+  /// Temporary helper to safely get length from ExtendedValue until
+  /// migration of all character manipulation to CharBoxValue/CharArrayBoxValue.
+  mlir::Value getLen(const fir::ExtendedValue &);
```

Comment:
This is redundant with `fir::getLen()` which is included in BoxValue.h, I believe.

Reply:
Is there one operating on `fir::ExtendedValue` ? They are member functions under CharBoxValue and CharArrayBoxValue, but I did not saw a generic one. Anyway, this one is a bit difference since it may open box, so it needs a builder in case we have and UnboxedValue boxchar that has no parent op accessible (I think we should avoid that, but right now that happens quite often).

Reply:
https://github.com/flang-compiler/f18-llvm-project/blob/fir-dev/flang/include/flang/Lower/Support/BoxValue.h#L214

Reply:
Thanks, I missed it. I removed the getLen I was adding since the cleanup get rid of the UnboxedValue case.